app/src/main/java/com/teamagam/gimelgimel/app/map/actions/timeplay/BetterPickersPicker.java

```diff
@@ -0,0 +1,70 @@
+package com.teamagam.gimelgimel.app.map.actions.timeplay;
+
+import android.content.Context;
+import android.support.v4.app.FragmentManager;
+import android.support.v7.app.AlertDialog;
+import com.codetroopers.betterpickers.calendardatepicker.CalendarDatePickerDialogFragment;
+import com.codetroopers.betterpickers.radialtimepicker.RadialTimePickerDialogFragment;
+import com.teamagam.gimelgimel.R;
+import com.teamagam.gimelgimel.domain.timeplay.DatePickerOpener;
+import java.util.Calendar;
+
+public class BetterPickersPicker
+    implements DatePickerOpener, CalendarDatePickerDialogFragment.OnDateSetListener {
+
+  private Context mContext;
+  private FragmentManager mFragmentManager;
+  private TimeplayViewModel.TextTimeDisplayer mTextTimeDisplayer;
+
+  public BetterPickersPicker(Context context, FragmentManager fragmentManager) {
+    mContext = context;
+    mFragmentManager = fragmentManager;
+  }
+
+  public void setOnDateSetListener(TimeplayViewModel.TextTimeDisplayer textTimeDisplayer) {
```

Comment:
I'd expect a set listener method to accept a listener and not a displayer

Reply:
And it would be probably better to rename it to setOnDateSelectedListener

Reply:
done